Rename "node" to "signature" (Fix of Issue #138)

dongbohu opened the pull request:
This PR fixes issue #138. It affects both back end and front end files. 

After talking to Rene, I moved the functionalities in migration file `0009_auto_20170503_1700.py` to deployment in `fabfile/adage_server.py`. 

@mhuyck, please review **at least** the following front end files carefully to make sure that my revisions are correct:
```
interface/src/app/analyze/analysis/analysis.js
interface/src/app/analyze/analysis/sampleBin.js
interface/src/app/volcano-plot/volcano-plot.js
```

This file is also revised: `interface/src/app/help/help.tpl.html`, but I am not sure whether we should keep both "Node" and "Signature" sections. It is confusing to me.

rzelayafavila commented on adage/analyze/migrations/0009_auto_20170503_1700.py:
```diff
@@ -1,28 +1,17 @@
+
+"""This migration file used to generate a default participation type and
+set all Participation records to the default type.  These operations
+have been moved to fabfile/adage_server.py now.
```
:+1:

dongbohu replied:
That is a good point. I will remove the `create_participation_type_if_nonexistent` function definition completely.

rzelayafavila commented on fabfile/adage_server.py:
```diff
@@ -187,6 +187,16 @@ def import_data_and_index():
+        "underlying process or processes captured by the signature."
+    )
+    run('python manage.py create_or_update_participation_type "%s" "%s"'
+        % (participation_type_name, participation_type_desc))
```
:+1:

dongbohu commented on adage/analyze/management/commands/import_activity.py:
```diff
@@ -65,58 +65,61 @@ def import_activity(file_handle, ml_model_name):
+def import_signatures(signatures, mlmodel):
     """
-    Load input nodes into "Node" table in the database.
+    Load input signatures into "Signaturee" table in the database.
```
Good catch! Thanks.

mhuyck commented on interface/src/app/help/help.tpl.html:
```diff
@@ -88,12 +88,12 @@ <h3><u>Signatures</u></h3>
+    different signatures have different numbers of high weight genes. High
+    weight genes are used to characterize a signature because their expression
+    levels have the largest influence on the activity of the signature. A
+    signature is composed of high weight genes on one side of a node.
```
My understanding: "node"s are the abstract term used in the denoising autoencoder literature, and "signatures" refer to our interpretation of those nodes in the context of the _Pseudomonas_ ArrayExpress gene expression data in the ADAGE model.

mhuyck commented:
@dongbohu: do you have a test server where this is running?

We should also test deployment, since this code alters how that works. Have you done that?

cgreene commented on interface/src/app/help/help.tpl.html:
```diff
@@ -88,12 +88,12 @@ <h3><u>Signatures</u></h3>
   <div class="lead" id="High Weight Gene">
     <h3><u>High Weight Gene</u></h3>
```
How about:

> The ADAGE model is made up of features (also known as nodes). For each feature, every gene has a positive or negative weight. A signature is composed of the genes on one side (positive or negative) of a feature. The genes that are termed high weight are those that are 2.5 or more standard deviations from the mean gene weight. These high weight genes are used to characterize each signature because their expression levels have the largest influence on the activity of the signature.

dongbohu replied:
@casey, do you have any comments on the two paragraphs that describe `Nodes` (lines 61-66) and `Signatures` (lines 72-84) in `help.tpl.html`?

dongbohu commented on interface/src/app/analyze/analysis/sampleBin.js:
```diff
@@ -325,7 +325,7 @@ MathFuncts, errGen) {
         .posKey('activity')
         .data(this.getSampleObjects());
-      this.heatmapData.samples = sampleClust.orderedNodes().map(
+      this.heatmapData.samples = sampleClust.orderedSignatures().map(
```
Thanks. That is why I really want to borrow your sight for this file!

dongbohu commented:
I believe I have addressed all comments from @mhuyck and @cgreene.

@mhuyck: So far I only tested the front end and a few management commands. A deployment test will be great since this PR affects a lot of deployment related files. Can you do that when you get time?

mhuyck commented:
Deployment succeeded... test server up and running at http://greenelab.fgtech.com/ . I'll do a final review of your changes in the morning, but I think we're in good shape!

dongbohu commented:
Thanks @mhuyck! Glad that there is no deployment issues.

mhuyck commented on interface/src/app/help/help.tpl.html:
```diff
@@ -87,13 +87,13 @@ <h3><u>Signatures</u></h3>
+    The genes that are termed high weight are those that are 2.5 or more
+    standard deviations from the mean gene weight. These high weight genes
+    are used to characterize each signature because their expression levels
+    have the largest influence on the activity of the signature.
```
👍

mhuyck commented:
Looks like you got everything. Thank you so much for fixing this, @dongbohu!

mhuyck commented:
Let me just say, for the record, that I don't think much of codeclimate's gripes about "cyclomatic complexity." The code in `import_signatures` is very clear.

The test code in `create_participations` is also quite readable with little to be gained from splitting it up. Safe to ignore codeclimate there as well, in my opinion.
